# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [49]:
#prueba de hipótesis para comparar si hay 
#diferencias significativas entre dos 
#dietas (prueba 1)

#hipotesis nula: No existe 
# una diferencia significativa en la 
# pérdida de peso promedio entre las personas 
# que siguieron
#  la Dieta 1 y las que siguieron la Dieta 2.
#Hipotesis alternativa:
# Sí existe una diferencia 
# significativa en la pérdida de peso 
# promedio entre las personas que siguieron 
# la Dieta 1 y las que siguieron la Dieta 2.
import pandas as pd
import numpy as np
import scipy.stats as stats
diet1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]
diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# Student's t-test
t_value, p_stat = stats.ttest_ind(diet_1, diet_2)
print(f"t-value: {t_value:.3f}")
print(f"p-value: {p_stat:.3f}")

diets = pd.DataFrame({
    "Diet 1": diet1,
    "Diet 2": diet2
})

"""diets2 = pd.DataFrame(data=list(zip(diet1, diet2)),
                      columns=["Diet 1", "Diet 2"])
"""
resumen = pd.DataFrame({
    "Medias": diets.mean(),
    "Desviaciones": diets.std(),
    "Mínimos": diets.min(),
    "Máximos": diets.max()
})

t_stat, p_value = stats.ttest_ind(diets['Diet 1'], diets['Diet 2'], equal_var=False)


print(f'La media de cada dieta es de {diets.mean()}')
print("Resumen estadístico de las dietas:")
print(resumen)
print(f"t-statistic: {t_stat:.3f}")
print(f"p-valor: {p_value:.3f}")



t-value: -3.538
p-value: 0.004
La media de cada dieta es de Diet 1    2.542857
Diet 2    3.028571
dtype: float64
Resumen estadístico de las dietas:
          Medias  Desviaciones  Mínimos  Máximos
Diet 1  2.542857      0.330944      2.0      3.0
Diet 2  3.028571      0.149603      2.8      3.2
t-statistic: -3.538
p-valor: 0.007


Con base en la prueba t de Student para muestras independientes (t = −3.538; p = 0.007 < 0.05), se rechaza la hipótesis nula y se concluye que existe una diferencia estadísticamente significativa en la pérdida de peso promedio entre ambas dietas.
En esta muestra, la Dieta 2 resultó más efectiva para la reducción de peso después de un mes.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

H0: No existen diferencias significativas en el crecimiento promedio de las plantas entre los tres fertilizantes.

H1: Al menos un fertilizante produce un crecimiento promedio significativamente diferente de los demás.

In [43]:
fertilizante1 = [20, 21, 20, 19, 20]
fertilizante2 = [22, 21, 23, 22, 21]
fertilizante3 = [24, 23, 22, 23, 24]

F_stat, p_value = stats.f_oneway(fertilizante1, fertilizante2, fertilizante3)

print(f"F: {F_stat:.3f}")
print(f"p-valor: {p_value:.3f}")

F: 20.316
p-valor: 0.000


Con un nivel de significación del 5%, la prueba ANOVA de un factor muestra un estadístico F = 20.316 y un p-valor = 0.000.
Dado que el p-valor < 0.05, se rechaza la hipótesis nula y se concluye que al menos uno de los fertilizantes genera un crecimiento promedio significativamente diferente.

In [54]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
data = [20,21,20,19,20, 22,21,23,22,21, 24,23,22,23,24]
grupos = ['F1']*5 + ['F2']*5 + ['F3']*5

df = pd.DataFrame({'Crecimiento': data, 'Fertilizante': grupos})

tukey = pairwise_tukeyhsd(endog=df['Crecimiento'], groups=df['Fertilizante'], alpha=0.05)
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
    F1     F2      1.8 0.0099 0.4572 3.1428   True
    F1     F3      3.2 0.0001 1.8572 4.5428   True
    F2     F3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


Con un nivel de significación del 5% (α = 0.05), los resultados de la prueba de Tukey HSD muestran diferencias estadísticamente significativas en el crecimiento promedio entre los tres fertilizantes.

En particular:

F2 produce un crecimiento mayor que F1.

F3 produce un crecimiento mayor que F1 y F2.

Por lo tanto, el fertilizante 3 es significativamente más eficaz para el crecimiento de las plantas que los otros dos.